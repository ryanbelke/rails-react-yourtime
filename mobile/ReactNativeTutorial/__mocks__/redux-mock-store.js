import configureMockStore from 'redux-mock-store';
import { initialState as reduxInitialState } from 'ReactNativeTutorial/app/reducers';

import mockThunkMiddleware from './mockThunkMiddleware';

export const createStoreFromState = configureMockStore([mockThunkMiddleware]);
export const initialState = reduxInitialState;

export default () => createStoreFromState(initialState);
