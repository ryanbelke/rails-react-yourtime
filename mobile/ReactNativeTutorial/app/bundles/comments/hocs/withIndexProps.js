/* eslint-disable react/no-unused-prop-types */
// @flow
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { createSelector } from 'reselect';

import commentsPropsSelector from 'ReactNativeTutorial/app/selectors/commentsPropsSelector';
import * as actions from '../thunks';

type CommentType = {
  author?: string,
  text?: string,
}

export type IndexPropsType = {
  comments: Array<CommentType>,
  meta: {
    loading: boolean,
  },
  actions: {
    fetch: () => void,
    updateForm: (payload: Object) => void,
    createComment: (payload: Object) => void,
  }
}

const mapStateToProps = createSelector(
  commentsPropsSelector,
  props => props.toJS()
);

const mapDispatchToProps = (dispatch: Function) => ({
  actions: bindActionCreators(actions, dispatch),
});

export default (Component: ReactClass<IndexPropsType>) =>
  connect(mapStateToProps, mapDispatchToProps)(Component);
