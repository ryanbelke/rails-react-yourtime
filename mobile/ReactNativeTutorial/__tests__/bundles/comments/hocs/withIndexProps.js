import React from 'react';
import withIndexProps from 'ReactNativeTutorial/app/bundles/comments/hocs/withIndexProps';
import Mock from 'mock';
import renderer from 'react-test-renderer';
import { createStoreFromState, initialState } from 'redux-mock-store';

describe('withIndexProps', () => {
  it('adds props to Index component', () => {
    const state = initialState.mergeDeep({
      commentsStore: {
        2: { id: 2, author: 'Justin', text: 'Another random comment' },
        3: { id: 3, author: 'John', text: 'Yet another random comment' },
        1: { id: 1, author: 'Alexey', text: 'Just some random comment' },
      },
    });
    const store = createStoreFromState(state);
    const Component = withIndexProps(Mock);
    const tree = renderer.create(<Component store={store} />);
    expect(tree).toMatchSnapshot();
  });
});
