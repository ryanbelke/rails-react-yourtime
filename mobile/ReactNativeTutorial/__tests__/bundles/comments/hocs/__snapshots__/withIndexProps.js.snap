exports[`withIndexProps adds props to Index component 1`] = `
<Mock
  actions={
    Object {
      "createComment": [Function],
      "fetch": [Function],
      "updateForm": [Function],
    }
  }
  comments={
    Array [
      Object {
        "author": "John",
        "id": 3,
        "text": "Yet another random comment",
      },
      Object {
        "author": "Justin",
        "id": 2,
        "text": "Another random comment",
      },
      Object {
        "author": "Alexey",
        "id": 1,
        "text": "Just some random comment",
      },
    ]
  }
  meta={
    Object {
      "loading": false,
    }
  } />
`;
